src/hub/PolicyHub.Service/Program.cs

```diff
@@ -31,20 +33,24 @@ await WebApplicationBuildRunner
     .BuildAndRunWebApplicationAsync<Program>(args, "policy-hub", version, ".Hub",
         builder =>
         {
-            builder.Services.AddTransient<IClaimsTransformation, KeycloakClaimsTransformation>();
-            builder.Services.AddEndpointsApiExplorer();
-            builder.Services.AddHubRepositories(builder.Configuration);
-            builder.Services.ConfigureHttpJsonOptions(options =>
-            {
-                options.SerializerOptions.Converters.Add(new JsonStringEnumConverter());
-            });
-            builder.Services.Configure<Microsoft.AspNetCore.Mvc.JsonOptions>(options =>
-            {
-                options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
-            });
+            builder.Services
+                .AddSingleton<IErrorMessageService, ErrorMessageService>()
+                .AddSingleton<IErrorMessageContainer, ErrorMessageContainer>()
+                .AddTransient<IClaimsTransformation, KeycloakClaimsTransformation>()
+                .AddEndpointsApiExplorer()
+                .AddHubRepositories(builder.Configuration)
+                .ConfigureHttpJsonOptions(options =>
+                    {
+                        options.SerializerOptions.Converters.Add(new JsonStringEnumConverter());
+                    })
+                .Configure<Microsoft.AspNetCore.Mvc.JsonOptions>(options =>
+                    {
+                        options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
+                    });
         },
         (app, _) =>
         {
+            app.UseMiddleware<GeneralHttpErrorHandler>();
```

Comment:
As this is on net9 we should use the IExceptionFilter as in https://github.com/eclipse-tractusx/portal-backend/pull/1272